core/src/main/scala/cats/Functor.scala

```diff
@@ -159,10 +159,16 @@ import simulacrum.typeclass
    * }}}
    *
    */
-  def unzip[A, X, Y](fa: F[A])(implicit ev: A <~< (X, Y)): (F[X], F[Y]) = {
+  //TODO: Drop from Cats 3.0
+  @noop
+  def unzip[A, B](fab: F[(A, B)]): (F[A], F[B]) = (map(fab)(_._1), map(fab)(_._2))
+
+  //TODO: Drop `final` when the `noop` version deprecates
+  // `final` restricts functor instance overriding of `unzip` to the noop version
+  final def unzip[A, X, Y](fa: F[A])(implicit ev: A <~< (X, Y)): (F[X], F[Y]) = {
     val fxy = map(fa)(ev)
 
-    (map(fxy)(_._1), map(fxy)(_._2))
+    unzip(fxy)
   }
```

Comment:
@travisbrown Overloading seems to work but I'm not sure if there are any negative implication.

Reply:
@yangzai I think it's a good idea to avoid overloading when it's not absolutely necessary—see e.g. [this SO answer](https://stackoverflow.com/a/2512001/334519) for some reasons. Not everything there is still applicable in the latest Scala versions, but some is. For example, this is a nicer error message:

```scala
scala> import cats.Functor, cats.instances.all._
import cats.Functor
import cats.instances.all._

scala> Functor[Option].unzip(1)
                             ^
       error: type mismatch;
        found   : Int(1)
        required: Option[(?, ?)]
```
…than this:
```scala
scala> Functor[Option].unzip(1)
<console>:16: error: overloaded method value unzip with alternatives:
  [A, X, Y](fa: Option[A])(implicit ev: A <~< (X, Y))(Option[X], Option[Y]) <and>
  [A, B](fab: Option[(A, B)])(Option[A], Option[B])
 cannot be applied to (Int)
       Functor[Option].unzip(1)
                       ^
```
If we _had_ to do this to get the syntax method, it'd be different, but I don't think overloading is worth it just to get Simulacrum to generate the syntax method. Implementing it explicitly in `FunctorOps` is also more consistent with the approach we use in other cases where Simulacrum hasn't done what we needed.

Reply:
@travisbrown Yeah it was a lazy hack. Sorry was kind of busy the past week. Will reference #3234 for the `FunctorOps` approach if the `<~<` based type signature is fine going forward.